local-archive-endpoint/server.js

```diff
@@ -0,0 +1,182 @@
+    return new Promise((resolve, reject) => {
+        console.log('Running typegen command...')
+        const typegenProcess = spawn('sqd', ['typegen:' + runtime], {
+            stdio: 'inherit',
+        })
+
+        typegenProcess.on('close', (code) => {
+            if (code === 0) {
+                console.log('Typegen command completed successfully')
+                resolve()
+            } else {
+                reject(new Error(`Typegen command failed with code ${code}`))
+            }
+        })
+    })
+}
+
+app.get('/', async (req, res) => {
+    const metadataFile = `./local-archive-endpoint/runtime.metadata`
+    const archiveURL = `https://v2.archive.subsquid.io/metadata/${runtime}`
+
+    try {
+        console.log(`Fetching original metadata from ${archiveURL}`)
+        const response = await axios.get(archiveURL)
+        const metadata = response.data
+        console.log('Fetched original metadata successfully')
+
+        const latestSpecVersion = getLastSpecVersion(metadata)
+        const newSpecVersion = latestSpecVersion + 1
+
+        const localRuntimeMetadata = readLocalMetadata(metadataFile)
+        const customMetadata = {
+            blockNumber: 0,
+            blockHash: '0x0000',
+            specName: runtime,
+            specVersion: newSpecVersion,
+            metadata: localRuntimeMetadata,
+        }
+
+        const concatMetadata = concatenateMetadata(metadata, customMetadata)
+
+        res.send(concatMetadata)
+    } catch (error) {
+        console.error(`Error fetching or processing metadata: ${error.message}`)
+        res.status(500).send('Error fetching or processing metadata')
+    }
+})
+
+async function run(runtimeParam) {
+    runtime = runtimeParam
+
+    try {
+        console.log('Extracting metadata with subwasm')
+        const subwasmCommand = `subwasm metadata local-archive-endpoint/${runtime}_runtime.compact.compressed.wasm -f hex+scale > local-archive-endpoint/runtime.metadata`
+        execSync(subwasmCommand, { stdio: 'inherit' })
+        console.log('Metadata extracted successfully')
+
+        const newSpecVersionUrl = `http://localhost:${port}`
+        const originalSpecVersionUrl = updateTypegenFile(
+            runtime,
+            newSpecVersionUrl
+        )
+
+        const server = await startServer()
+
+        try {
+            await runTypegenCommand(runtime)
+        } catch (error) {
+            console.error(`Error running typegen command: ${error.message}`)
+        } finally {
+            revertTypegenFile(runtime, originalSpecVersionUrl)
+            console.log('Shutting down server...')
+            server.close(() => {
+                process.exit(0)
+            })
+        }
+    } catch (error) {
+        console.error(`Error running subwasm or typegen: ${error.message}`)
+        process.exit(1)
+    }
+}
+
+const runtimeParam = process.argv[2]
+if (!runtimeParam) {
+    console.error('Usage: sqd pre-runtime-upgrade <runtime_name>')
```

Comment:
Maybe it's better to pass the path to the file instead of making assumptions about where it's stored and what it's called. 
```suggestion
    console.error('Usage: sqd pre-runtime-upgrade <runtime_name> <path_to_runtime>')
```